pkg/schedule/schedulers/balance_range.go

```diff
@@ -67,15 +74,102 @@ func (handler *balanceRangeSchedulerHandler) listConfig(w http.ResponseWriter, _
 	}
 }
 
+func (handler *balanceRangeSchedulerHandler) addJob(w http.ResponseWriter, r *http.Request) {
+	var input map[string]any
+	if err := apiutil.ReadJSONRespondError(handler.rd, w, r.Body, &input); err != nil {
+		return
+	}
+	job := &balanceRangeSchedulerJob{
+		Create:  time.Now(),
+		Status:  pending,
+		Timeout: time.Hour,
+	}
+	job.Engine = input["engine"].(string)
```

Comment:
if "engine" is not exist in input, it will be unsafe

Reply:
yes, I fixed it even if TIDB check it.